x-pack/plugins/data_visualizer/public/application/file_data_visualizer/components/import_view/import_view.js

```diff
@@ -86,12 +85,14 @@ export class ImportView extends Component {
 
   componentDidMount() {
     this.loadDataViewNames();
+    this.loadDefaultIndexSettings();
```

Comment:
because of the way the File Data Visualizer is embedded as a component in ML and Home, this seemed to be the simplest way for the plugin to determine which settings to use. This was based on my original implementation where the serverless plugins called the file upload plugin on the server side.
However, I was also unaware of being able to use `buildFlavor` which also works client side. So it looks like this PR can can simplified even more and this endpoint removed.